source/slang/slang-emit-glsl-extension-tracker.cpp

```diff
@@ -26,18 +26,35 @@ void GLSLExtensionTracker::requireVersion(ProfileVersion version)
     }
 }
 
-void GLSLExtensionTracker::requireHalfExtension()
+void GLSLExtensionTracker::requireBaseTypeExtension(BaseType baseType)
 {
-    if (!m_hasHalfExtension)
+    uint32_t bit = 1 << int(baseType);
+    if (m_hasBaseTypeFlags & bit)
```

Comment:
I suppose the micro-optimization here of not even trying to add the strings pays off for stuff like basic types, since we'd call into this logic a lot. Otherwise I'd point out that we already avoid re-adding an extension that was already added as part of `requireExtension`.